### DISCUSSION TITLE
func class attributes doesn't store the form of the line shape?

### DISCUSSION THREAD

**Discussion by bertinets** [lmfit/lmfit-py#835] [Q&A] [answered]
Hi,
I'm starting to use lmfit and I am not sure I fully understand what the func attribute of the class Model does for those builtin models that can be expressed in different forms (like StepModel, RectangleModel ThermalDistributionModel etc...). 

For all of those models, no matter what I pass as an argument for the 'form' keyword, when I try to call the 'func' attribute I get always the default form.
example:
```
import matplotlib.pyplot as plt
import numpy as np
from lmfit.models import StepModel
model = StepModel(form='logistic')
model.func
```
returns:
`<function lmfit.lineshapes.step(x, amplitude=1.0, center=0.0, sigma=1.0, form='linear')>`

of course, the model is generated properly, in fact, by calling `model._get_state()` returns 'logistic' for the keyword 'form', although in the function definition it shows 'linear':

```
(('step',
  <function lmfit.lineshapes.step(x, amplitude=1.0, center=0.0, sigma=1.0, form='linear')>,
  'step',
  '',
  ['x'],
  ['amplitude', 'center', 'sigma'],
  {},
  'raise',
  {'form': 'logistic'}),
 None,
 None)
```

The same seems to happen for the other models that accept different forms.

Is there a reason why the form of the class attribute func doesn't match the actual form of the line shape?

**Comment by newville** [comment, marked as answer]
@bertinets Yeah, this could probably be done better.    Then again,  there really is not an expectation that people would call `Model.func()` without constructing the function arguments, including the values in `opts`.

**Reply by bertinets** [reply]
Ok, thanks, that's what I thought. It just got me confused at the beginning and I was wondering if that could have affected in any way the functionality of the library.
